index.ts

```diff
@@ -396,36 +411,81 @@ export async function createElasticsearchMcpServer(
           ],
         };
       }
-    }
+    },
   );
 
   return server;
 }
 
 const config: ElasticsearchConfig = {
+  sse_addr: process.env.SSE_ADDR || "",
   url: process.env.ES_URL || "",
   apiKey: process.env.ES_API_KEY || "",
   username: process.env.ES_USERNAME || "",
   password: process.env.ES_PASSWORD || "",
   caCert: process.env.ES_CA_CERT || "",
 };
 
+async function create_sse_server(server: any) {
+  const app = express();
+
+  // Use morgan to log every request
+  app.use(morgan("combined"));
+
+  const transports: { [sessionId: string]: SSEServerTransport } = {};
+
+  app.get("/sse", async (_: Request, res: Response) => {
+    const transport = new SSEServerTransport("/messages", res);
+    transports[transport.sessionId] = transport;
+    res.on("close", () => {
+      delete transports[transport.sessionId];
+    });
+    await server.connect(transport);
+  });
+
+  app.post("/messages", async (req: Request, res: Response) => {
+    const sessionId = req.query.sessionId as string;
+    const transport = transports[sessionId];
+    if (transport) {
+      await transport.handlePostMessage(req, res);
+    } else {
+      res.status(400).send("No transport found for sessionId");
+    }
+  });
+
+  const sseAddr = process.env.SSE_ADDR || "127.0.0.1:3000";
+  const [host, port] = sseAddr.split(":");
+  if (!port) {
+    console.error("Invalid SSE_ADDR format. Expected 'host:port'.");
+    process.exit(1);
+  }
+
+  app.listen(Number(port), host, () => {
```

Comment:
Suggest we need a way to handle termination signals in SSE mode gracefully by stopping the SSE server properly and disconnect any active SSEServerTransport instances and clean up the transports object.